Neos.Neos/Classes/Domain/Service/WorkspaceService.php

```diff
@@ -0,0 +1,365 @@
+     * Create a new root (aka base) workspace with the specified metadata
+     */
+    public function createRootWorkspace(ContentRepositoryId $contentRepositoryId, WorkspaceTitle $title, WorkspaceDescription $description): WorkspaceName
+    {
+        $contentRepository = $this->contentRepositoryRegistry->get($contentRepositoryId);
+        $workspaceName = self::getUniqueWorkspaceName($contentRepository, $title->value);
+        $contentRepository->handle(
+            CreateRootWorkspace::create(
+                $workspaceName,
+                DeprecatedWorkspaceTitle::fromString($title->value),
+                DeprecatedWorkspaceDescription::fromString($description->value),
+                ContentStreamId::create()
+            )
+        );
+        $this->addWorkspaceMetadata($contentRepositoryId, $workspaceName, $title, $description, WorkspaceClassification::ROOT, null);
+        return $workspaceName;
+    }
+
+    /**
+     * Create a new, personal, workspace for the specified user
+     */
+    public function createPersonalWorkspace(ContentRepositoryId $contentRepositoryId, WorkspaceTitle $title, WorkspaceDescription $description, WorkspaceName $baseWorkspaceName, UserId $ownerId): WorkspaceName
+    {
+        return $this->createWorkspace($contentRepositoryId, $title, $description, $baseWorkspaceName, $ownerId, WorkspaceClassification::PERSONAL);
+    }
+
+    /**
+     * Create a new, potentially shared, workspace
+     */
+    public function createSharedWorkspace(ContentRepositoryId $contentRepositoryId, WorkspaceTitle $title, WorkspaceDescription $description, WorkspaceName $baseWorkspaceName, UserId|null $ownerId = null): WorkspaceName
+    {
+        return $this->createWorkspace($contentRepositoryId, $title, $description, $baseWorkspaceName, $ownerId, WorkspaceClassification::SHARED);
+    }
+
+    /**
+     * Create a new, personal, workspace for the specified user if none exists yet
+     */
+    public function createPersonalWorkspaceForUserIfMissing(ContentRepositoryId $contentRepositoryId, User $user): void
+    {
+        $existingWorkspaceName = $this->findPrimaryWorkspaceNameForUser($contentRepositoryId, $user->getId());
+        if ($existingWorkspaceName !== null) {
+            $this->requireWorkspace($contentRepositoryId, $existingWorkspaceName);
+            return;
+        }
+        $this->createPersonalWorkspace(
+            $contentRepositoryId,
+            WorkspaceTitle::fromString($user->getLabel()),
+            WorkspaceDescription::empty(),
+            WorkspaceName::forLive(),
+            $user->getId(),
+        );
+    }
+
+    /**
+     * Determines the permission the given user has for the specified workspace {@see WorkspacePermissions}
+     */
+    public function getWorkspacePermissionsForUser(ContentRepositoryId $contentRepositoryId, WorkspaceName $workspaceName, User $user): WorkspacePermissions
+    {
+        $userIsAdministrator = $this->userService->currentUserIsAdministrator();
```

Comment:
we should probably at some point have a different logic for that than to use the flow `Neos.Neos:Administrator` right? But maybe not either ... so doesnt matter... but i was thinking about a privilege target that the admin has and that you can add to your own `CRAdministrator` role.

Reply:
I don't understand